hkmc2/shared/src/main/scala/hkmc2/codegen/Block.scala

```diff
@@ -390,3 +390,28 @@ enum Value extends Path:
 
 case class Arg(spread: Bool, value: Path)
 
+extension (k: Block => Block)
+  
+  def chain(other: Block => Block): Block => Block = b => k(other(b))
+  def rest(b: Block): Block = k(b)
+  def transform(f: (Block => Block) => (Block => Block)) = f(k)
+  
+  def assign(l: Local, r: Result) = k.chain(Assign(l, r, _))
+  def assignFieldN(lhs: Path, nme: Tree.Ident, rhs: Result) = k.chain(AssignField(lhs, nme, rhs, _)(N))
+  def break(l: Local): Block = k.rest(Break(l))
+  def continue(l: Local): Block = k.rest(Continue(l))
+  def define(defn: Defn) = k.chain(Define(defn, _))
+  def end = k.rest(End())
+  def ifthen(scrut: Path, cse: Case, trm: Block): Block => Block = k.chain(Match(scrut, cse -> trm :: Nil, N, _))
+  def label(label: Local, body: Block) = k.chain(Label(label, body, _))
+  def ret(r: Result) = k.rest(Return(r, false))
+  def staticif(b: Boolean, f: (Block => Block) => (Block => Block)) = if b then k.transform(f) else k
+
+def blockBuilder: Block => Block = identity
+
+extension (p: Path)
+  def selN(id: Tree.Ident) = Select(p, id)(N)
+  def asArg = Arg(false, p)
+
+extension (l: Local)
+  def asPath: Path = Value.Ref(l)
```

Comment:
could this be methods in the class directly instead of extension method

Reply:
Agreed.

Reply:
Good catch, thanks